i18n-helpers/src/bin/mdbook-gettext.rs

```diff
@@ -25,111 +25,61 @@
 
 use anyhow::{anyhow, Context};
-use mdbook::book::Book;
 use mdbook::preprocess::{CmdPreprocessor, PreprocessorContext};
-use mdbook::BookItem;
-use mdbook_i18n_helpers::{extract_events, reconstruct_markdown, translate_events};
+use mdbook_i18n_helpers::gettext::{add_stripped_summary_translations, translate_book};
 use polib::catalog::Catalog;
-use polib::message::Message;
 use polib::po_file;
-use pulldown_cmark::Event;
 use semver::{Version, VersionReq};
+use std::path::PathBuf;
 use std::{io, process};
 
-/// Strip formatting from a Markdown string.
+/// Check whether the book should be transalted.
 ///
-/// The string can only contain inline text. Formatting such as
-/// emphasis and strong emphasis is removed.
-///
-/// Modelled after `mdbook::summary::stringify_events`.
-fn strip_formatting(text: &str) -> String {
-    extract_events(text, None)
-        .iter()
-        .filter_map(|(_, event)| match event {
-            Event::Text(text) | Event::Code(text) => Some(text.as_ref()),
-            Event::SoftBreak => Some(" "),
-            _ => None,
-        })
-        .collect()
-}
-
-fn translate(text: &str, catalog: &Catalog) -> String {
-    let events = extract_events(text, None);
-    let translated_events = translate_events(&events, catalog);
-    let (translated, _) = reconstruct_markdown(&translated_events, None);
-    translated
-}
-
-/// Update `catalog` with stripped messages from `SUMMARY.md`.
-///
-/// While it is permissible to include formatting in the `SUMMARY.md`
-/// file, `mdbook` will strip it out when rendering the book. It will
-/// also strip formatting when sending the book to preprocessors.
-///
-/// To be able to find the translations for the `SUMMARY.md` file, we
-/// append versions of these messages stripped of formatting.
-fn add_stripped_summary_translations(catalog: &mut Catalog) {
-    let mut stripped_messages = Vec::new();
-    for msg in catalog.messages() {
-        // The `SUMMARY.md` filename is fixed, but we cannot assume
-        // that the file is at `src/SUMMARY.md` since the `src/`
-        // directory can be configured.
-        if !msg.source().contains("SUMMARY.md") {
-            continue;
-        }
-
-        let message = Message::build_singular()
-            .with_msgid(strip_formatting(msg.msgid()))
-            .with_msgstr(strip_formatting(msg.msgstr().unwrap()))
-            .done();
-        stripped_messages.push(message);
+/// The book should be translated if:
+/// * `book.language` is defined in mdbook config
+/// * Corresponding {language}.po defined
+fn should_translate(ctx: &PreprocessorContext) -> anyhow::Result<bool> {
+    // Translation is a no-op when the target language is not set
+    if ctx.config.book.language.is_none() {
+        return Ok(false);
     }
 
-    for msg in stripped_messages {
-        catalog.append_or_update(msg);
-    }
+    let path = get_catalog_path(ctx)?;
+    // Nothing to do if PO file is missing.
+    return Ok(path.try_exists()?);
```

Comment:
I think this is the same as `path.try_exists()` — perhaps with a `.into()` or so tacked on?

You should at least trip the `return` which is not necessary here.

Reply:
Ah, looking again, I think we could get away with letting this function return `bool` directly — and then you use a [`is_ok`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.is_ok) to convert the `Result` here. That simplified the other functions a tiny bit too.

Reply:
Sounds good, I flattened the type and now it returns just `bool`.